### PR TITLE
Handle jshinting via ember-cli's new 'lint' hooks

### PR DESCRIPTION
**Pull request by ef4**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -80,5 +83,16 @@ module.exports = {
 
   _readTemplate: function(name) {
     return fs.readFileSync(path.join(__dirname, 'templates', name + '.html'));
+  },
+
+  postprocessTree: function(type, tree) {
+    if (type === 'lint-app' || type === 'lint-tests' || type === 'lint-addon') {
+      return jshintTrees(tree, {
+        jshintrcPath: this.jshintrc.tests,
+        description: 'JSHint - QUnit'
```
Comment (rwjblue) [inline index.js]: Can this be `'JSHint - ' + type` (otherwise they would all be reported as `JSHint - QUnit`).

**Comment by rwjblue** [comment]
One small nit-pick, but LGTM.

File: index.js (hunk 1 above)
Comment (ef4) [inline index.js]: Well, these are all QUnit specific. But I could also add the type.

Comment (rwjblue) [inline index.js]: Yes, the point is the type. When this shows up in the slow trees log how do we know WHICH tree was slow?

Comment (ef4) [inline index.js]: Done.

**Comment by rwjblue** [comment]
@jakecraige - Can you release (and potentially add me to npm)?

**Comment by rwjblue** [comment]
@dgeb - Will need to be mirrored in ember-cli-mocha.

**Comment by ef4** [comment]
@dgeb I have a PR, I wasn't going to bother you until ember-cli releases.

**Comment by ef4** [comment]
Apologies, but I found a problem with the way I originally did this. We want linters to run in parallel, not in series, so we can't use the existing postprocessTree hook. Please take https://github.com/ember-cli/ember-cli-qunit/pull/29 to switch.

**Comment by dgeb** [comment]
@ef4 should I hold off on merging https://github.com/switchfly/ember-cli-mocha/pull/26 ?

**Comment by ef4** [comment]
switchfly/ember-cli-mocha#26 is already updated to match my change above, but you may want to wait until the corresponding change lands in ember-cli:

https://github.com/ember-cli/ember-cli/pull/3094

In case anybody else points out another reason the API would need to change.

**Pull request merged**